### ISSUE TITLE
LaTeX template not respected when YAML option rmd_subdir is used with non-boolean value

### ISSUE THREAD

**Issue by lorenzwalthert** [rstudio/bookdown#600]
Thanks for this package. I tried to use the rmd_subdir option introduced in #561. However, I encounter the following problem: 
When I specify a latex template (I took the default pandoc template) for the book, I lose it as soon as I use the option `rmd_subdir: ["rmd"]`. I forked and modified the [bookdown demo](https://github.com/rstudio/bookdown-demo) to create a reprex, which you can find [here](https://github.com/lorenzwalthert/bookdown-demo). Just see the last few commits. See https://github.com/lorenzwalthert/bookdown-demo/commit/d9bcee01fe7a78c505f90d618dda8a832858d311#diff-b5352b26567fa5f42ad7241975ba203bR6 for the commit where I added the option. If you remove the line and re-build the book, you will see that the template again takes effect. Otherwise the option seems to work fine. I don't understand what's going on. Any help appreciated.

**Comment by lorenzwalthert** [comment]
Also, when I use `rmd_subdir: true` I don't get the problem.

**Comment by lorenzwalthert** [comment]
cc: @ellisvalentiner

**Comment by ellisvalentiner** [comment]
@lorenzwalthert thanks for bringing this issue to my attention. I want to make sure I understand your problem.

As I understand your issue the problem is that the LaTeX template is not applied when you use the `rmd_subdir` config option with a list of subdirectories (e.g. `rmd_subdir: "rmd/"`).

Here is my understanding of a few scenarios:

1. if you are rendering book to PDF with template
a. without Rmd files in subdirectory (they are all top-level), then the template (tex/template.tex) is applied to the book correctly
b. with Rmd files in subdirectory, using `rmd_subdir: true` in `_bookdown.yml` then the template is applied to the book correctly
c. with Rmd files in subdirectory using `rmd_subdir: ["rmd/"]`  in `_bookdown.yml` then the template is NOT applied to the book (unexpected behavior)

Do I understand this correctly?

Would you also mind making the template a very obvious change? I didn't notice a difference in the output in any of these cases but it may have been too subtle for me. Can you tell me what to look for to know the template is being used?

**Comment by lorenzwalthert** [comment]
Just tried to reproduce the problem on my Mac and I can't in the sense that I get a LaTeX error when trying `rmd_subdir: true`-.- Will revert tomorrow to confirm your understanding, not 100% sure about 1a but I think you got it right.

**Comment by ellisvalentiner** [comment]
I am seeing two problems that may or not be related to your problem.

1. if `rmd_subdir` is a list of subdirectories and does not contain the project root (i.e. `"."`) then it misses the `index.Rmd` file.

2. but if `rmd_subdir` is a list of subdirectories and does contain the project root then Rmd files may be found more than once (because `list.files` is called with `recursive=TRUE`).

The first is a problem because the Pandoc-related configurations must be defined in the YAML metadata of the first Rmd file of the book for them to take effect. If we do not have the `index.Rmd` then these configurations (such as the LaTeX template) will not be applied.

The second is a problem because things like figures need unique labels in LaTeX.

My solution was to patch bookdown by making sure the project root is included in the source files and adding `unique(normalizePath(files))` in R/utils.R line 79.

However I have now encountered another problem rendering the reprex:

```
==> rmarkdown::render_site(output_format = 'bookdown::pdf_book', encoding = 'UTF-8')

./index.Rmd./rmd/01-intro.Rmd./rmd/02-literature.Rmd./rmd/03-method.Rmd./rmd/04-application.Rmd./rmd/06-references.Rmd./rmd2/05-summary.Rmd


processing file: bookdown-demo.Rmd
  |.......                                                          |  11%
   inline R code fragments

  |..............                                                   |  22%
label: unnamed-chunk-1 (with options) 
List of 1
 $ eval: logi FALSE

  |......................                                           |  33%
  ordinary text without R code

  |.............................                                    |  44%
label: unnamed-chunk-2 (with options) 
List of 1
 $ include: logi FALSE

  |....................................                             |  56%
  ordinary text without R code

  |...........................................                      |  67%
label: nice-fig (with options) 
List of 4
 $ fig.cap  : chr "Here is a nice figure!"
 $ out.width: chr "80%"
 $ fig.asp  : num 0.75
 $ fig.align: chr "center"

  |...................................................              |  78%
  ordinary text without R code

  |..........................................................       |  89%
label: nice-tab (with options) 
List of 1
 $ tidy: logi FALSE

  |.................................................................| 100%
   inline R code fragments


/usr/local/bin/pandoc +RTS -K512m -RTS bookdown-demo.utf8.md --to latex --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output bookdown-demo.tex --table-of-contents --toc-depth 2 --template tex/template.tex --number-sections --highlight-style tango --pdf-engine xelatex --natbib --include-in-header preamble.tex --variable tables=yes --standalone 
output file: bookdown-demo.knit.md

! Undefined control sequence.
l.165 {\centering \includegraphics
                                  [width=0.8\linewidth]{bookdown-demo_files/...

Error: Failed to compile bookdown-demo.tex. See bookdown-demo.log for more info.
Execution halted

Exited with status 1.
```

This problem seems to be that there is a directory mismatch - the bookdown files are in the project root and not in `_bookdown_files`. I have this problem only when I specify a LaTeX template option (i.e. `template: tex/template.tex`) and using either `rmd_subdir: true` or `rmd_subdir: "rmd"`.

**Comment by lorenzwalthert** [comment]
I am just seeing exactly this error.

**Comment by ellisvalentiner** [comment]
I don't think there is a best practice or recommendation yet.

- Putting `index.Rmd` into a subdirectory does not work for me because render function expects the file to be in the project root.
- Adding `index.Rmd` to `rmd_files` does not work for me either (it is missing the subdirectory files).
- Adding `"."` to `rmd_subdir` does work but I'm not sure whether users would just expect it to be included by default.

So I think the tasks are:

- [ ] decide whether users should have to explicitly include the project root when using `rmd_subdir` or if the project root should be implicitly included in the source files
- [ ] fix `source_files` to include the project root or `index.Rmd` (if applicable)
- [ ] figure out what is causing the problem with LaTeX directories above

**Comment by lorenzwalthert** [comment]
So my idea was just to leave `index.Rmd` in the project root and adding all files from the `rmd_subdir:` to the list of files to include and having some YAML-option combination that makes that work.

**Comment by lorenzwalthert** [comment]
I am back at work and the problem is that with my working example (which I can't disclose here) I have the problem described at the very top. However, in the reprex posted above, the error you [reported](https://github.com/rstudio/bookdown/issues/600#issuecomment-404622907) occurs now also on my machine. I added a commit in my [reprex](https://github.com/lorenzwalthert/bookdown-demo) to simply remove the code causing this error. The error most likely just occurred because the template did not define a controls sequence I used (i.e. the plotting), which is strange but anyways. So now it should be a reprex for the problem described in this issue. Sorry for the confusion. 
So I think now we are at the same page: 
> This problem seems to be that there is a directory mismatch - the bookdown files are in the project root and not in _bookdown_files. I have this problem only when I specify a LaTeX template option (i.e. template: tex/template.tex) and using either rmd_subdir: true or rmd_subdir: "rmd".

**Comment by ellisvalentiner** [comment]
Thanks for being patient. After some sleep and coffee I understand the code much better.

The problem is simply that `index.Rmd` is not the first file - therefore the `template` option is not applied.

Fortunately this is fairly easy to fix.

Will you install my [development branch](https://github.com/ellisvalentiner/bookdown/tree/issue-600) and tell me if it works for you?

**Comment by lorenzwalthert** [comment]
No worries, I don't expect other people to problems I face, but if it happens, I am obviously happy:-). I just tried your branch and it seems to work. Thanks a lot. 
***
cc: @n-gnecco 
TLDR: If you have Rmd files in the root that should not be part of the book, put the once you want to render into `rmd` and add 
```r
rmd_subdir: ["rmd"]
rmd_files: ["index.Rmd"]
``` 


to `_bookdown.yml`. You need the corresponding devel branch for this to work: 
```r
remotes::install_github("ellisvalentiner/bookdown@issue-600")
```

**Comment by yihui** [comment]
I just tweaked the PR. Could you test it?

```r
remotes::install_github("rstudio/bookdown#601")
```

You should not need `rmd_files: ["index.Rmd"]` any more.

**Comment by ellisvalentiner** [comment]
Your changes look good to me.

**Comment by yihui** [comment]
Thanks @ellisvalentiner! I'll merge the PR then. Please feel free to let me know if any of you find any problems.

**Comment by lorenzwalthert** [comment]
I just tested it and it works for my use case. Thanks @yihui  and @ellisvalentiner for your support.
